core/platform/auth/firebase_auth_services.py

```diff
@@ -71,78 +72,58 @@
 transaction_services = models.Registry.import_transaction_services()
 
 
-@contextlib.contextmanager
-def _acquire_firebase_context():
-    """Returns a context for calling the Firebase Admin SDK."""
-    app = firebase_admin.initialize_app()
-    try:
-        yield
-    finally:
-        if app is not None:
-            firebase_admin.delete_app(app)
+def establish_auth_session(request, response):
+    """Sets login cookies to maintain a user's sign-in session.
 
+    Args:
+        request: webapp2.Request. The request with the authorization to begin a
+            new session.
+        response: webapp2.Response. The response to establish the new session
+            upon.
+    """
+    claims = _get_auth_claims_from_session_cookie(_get_session_cookie(request))
 
-def _verify_id_token(auth_header):
-    """Verifies whether the auth_header has a valid Firebase-provided ID token.
+    if claims is not None:
+        return
 
-    Oppia's authorization headers use OAuth 2.0's Bearer authentication scheme.
+    token = _get_id_token(request)
+    claims = _get_auth_claims_from_id_token(token)
 
-    Bearer authentication (a.k.a. token authentication) is an HTTP
-    authentication scheme based on "bearer tokens", an encrypted JWT generated
-    by a trusted identity provider in response to login requests.
+    if claims is None:
+        return
```

Comment:
I think the confusion I have is this: the function purports to establish an auth session, but in the 2nd case it looks like it can exit without doing so. Shouldn't that lead to some kind of exception rather than a silent return (which conveys "all is well")?

Reply:
OK, since this seems to boil down to the same question in the other thread, I'll mark this one as resolved since the comments explain the intent (less comments makes it easier for me to address changes).